### DISCUSSION TITLE
How to setup sensitive column or handle sensitive filtering MySQL

### DISCUSSION THREAD

**Discussion by Long182k** [prisma/prisma#20845] [Q&A]
### Question

how I can setup sensitive filtering in my findMany() function.
```js
if (query.orderNumbers) {
console.log('query.orderNumbers', query.orderNumbers)
if (query.orderNumbers instanceof Array) {
where.AND.push({
orderNumber: {
in: query.orderNumbers,
},
})
} else {
where.AND.push({
orderNumber: query.orderNumbers as string,
})
}
}
```
### Expected behavior (optional)

1. Setup sensitive column in Prisma with MYSQL
2. Filtering sensitively with FindMany() 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Debian (Ubuntu)
- Database: MySQL
- Node.js version: 16.15

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Long182k 👋 

It looks like you're building a dynamic query based on certain conditions. You can extend this logic to include sensitive filtering.
```ts
import { Prisma } from '@prisma/client';

const query: Prisma.OrderFindManyArgs = {
  where: {
    AND: [], // Your existing conditions here
  },
};

if (query.orderNumbers) {
  if (query.orderNumbers instanceof Array) {
    query.where.AND.push({
      orderNumber: {
        in: query.orderNumbers,
      },
    });
  } else {
    query.where.AND.push({
      orderNumber: query.orderNumbers as string,
    });
  }
}

// Add sensitive filtering
if (query.sensitiveFilter) {
  query.where.AND.push({
    orderSecret: query.sensitiveFilter as string,
  });
}

const orders = await prisma.order.findMany(query);
```

In the code above, the `sensitiveFilter` parameter is assumed to be passed in the query object. When you want to filter sensitive data, you would add this parameter to your query object.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Long182k** [reply]
Hi @ludralph
Thanks for your reply. Could you talk more about sensitiveFilter parameter in the query object? Do you mean that each orderNumber also has an orderSecret and I will filter one more time by orderSecret to get the sensitive order ? 
My expectation is if I find out "apple" I just wanna receive "apple" not "apple" and "Apple" or add 1 more field "secret" to detect exactly "apple".

**Comment by Long182k** [comment]
Hi @Raphael Etim
Thanks for your reply. Could you talk more about sensitiveFilter parameter
in the query object? Do you mean that each orderNumber also has an
orderSecret and I will filter one more time by orderSecret to get the
sensitive order ?
My expectation is if I find out "apple" I just wanna receive "apple" not
"apple" and "Apple" or add 1 more field "secret" to detect exactly "apple".

Vào Th 6, 25 thg 8, 2023 vào lúc 21:37 Raphael Etim <
***@***.***> đã viết:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
